pkg/controller/complianceremediation/complianceremediation_controller.go

```diff
@@ -369,6 +379,51 @@ func (r *ReconcileComplianceRemediation) countKubeUnmetDependencies(rem *compv1a
 	return nMissingDeps, nil
 }
 
+func (r *ReconcileComplianceRemediation) handleUnsetValues(rem *compv1alpha1.ComplianceRemediation, logger logr.Logger) (reconcile.Result, error) {
+	_, hasUnsetXccdValues := rem.Annotations[compv1alpha1.RemediationUnsetValueAnnotation]
+
+	var nNotSetValues int
+	if hasUnsetXccdValues {
+		var xccdfValueErr error
+		nNotSetValues, xccdfValueErr = r.countXCCDFUnsetValues(rem, logger)
+		if xccdfValueErr != nil {
+			return reconcile.Result{}, xccdfValueErr
+		}
+	} else {
+		return reconcile.Result{}, fmt.Errorf("Remediation has RemediationUnsetValueAnnotation but no un-set XCCDF values found")
+	}
+
+	rCopy := rem.DeepCopy()
+	labels := rCopy.GetLabels()
+	if labels == nil {
+		labels = make(map[string]string)
+	}
+
+	if nNotSetValues > 0 {
+		if _, ok := labels[compv1alpha1.RemediationUnsetValueLabel]; !ok {
```

Comment:
In this case that remediations are parsed and have unset variables initially; suppose then the admin sets the variable and applies the remediation, would this label and RemediationUnsetValueAnnotation persist?

Reply:
This one is tricky, if the admin did set a variable value later on, should we trigger a rescan instead for that specify rule/might be more than one rule, however in order to do that we have to first associate all the variable with rules , and then a new remediation will be generated based on the new variable.

Reply:
Hmm, I don't think we should worry about triggering a rescan right away when the variable is set, but I want to let Jakub or Ozz weigh in on the handling of this part. My thinking of the user flow would be:
1. Scan + remediate attempt
2. Discover one or more remediation needs attention, from remediation status
3. Look at remediation and find the required variable, and set the value
4. re-run scan manually (or wait for the next scheduled scan)
The result of the second scan is that the remediations are updated with the variable setting.

Reply:
As far as the user flow is concerned, the user would need to modify the TailoredProfile in order to set the variable. While most variables are already set by default, this case is for variables that are empty and thus need setting. @Vincent056 perhaps in the PR description you could add an example of one of these variables that need setting? Anyway, the workflow that you described sounds correct.

Reply:
I think that's correct, there is one more case, when there is an value-required annotation, and the user did not set values for those value-required variables, it will also trigger that Need-Review Status.  I will add examples for that.